src/deprecated.jl

```diff
@@ -9,15 +9,15 @@
 )
 
 # TODO: remove tests when removed
-@deprecate function kerneldiagmatrix(κ::Kernel, X; obsdim::Int=2)
-    return kernelmatrix_diag(κ, X; obsdim=obsdim)
-end
-@deprecate function kerneldiagmatrix(κ::Kernel, X, Y; obsdim::Int=2)
-    return kernelmatrix_diag(κ, X, Y; obsdim=obsdim)
-end
-@deprecate function kerneldiagmatrix!(K::AbstractVector, κ::Kernel, X; obsdim::Int=2)
-    return kernelmatrix_diag!(K, κ, X; obsdim=obsdim)
-end
-@deprecate function kerneldiagmatrix!(K::AbstractVector, κ::Kernel, X, Y; obsdim::Int=2)
-    return kernelmatrix_diag!(K, κ, X, Y; obsdim=obsdim)
-end
+@deprecate kerneldiagmatrix(κ::Kernel, X; obsdim::Int=2) kernelmatrix_diag(
+    κ, X; obsdim=obsdim
+)
+@deprecate kerneldiagmatrix(κ::Kernel, X, Y; obsdim::Int=2) kernelmatrix_diag(
+    κ, X, Y; obsdim=obsdim
+)
+@deprecate kerneldiagmatrix!(K::AbstractVector, κ::Kernel, X; obsdim::Int=2) kernelmatrix_diag!(
+    K, κ, X; obsdim=obsdim
+)
+@deprecate kerneldiagmatrix!(K::AbstractVector, κ::Kernel, X, Y; obsdim::Int=2) kernelmatrix_diag!(
+    K, κ, X, Y; obsdim=obsdim
+)
```

Comment:
As mentioned earlier, this is not correct. Instead you have to use:
```suggestion
@deprecate kerneldiagmatrix(k::Kernel, x::AbstractVector) kernelmatrix_diag(k, x)
@deprecate kerneldiagmatrix(k::Kernel, x::AbstractVector, Y::AbstractVector) kernelmatrix_diag(k, x, y)
@deprecate kerneldiagmatrix!(out::AbstractVector, k::Kernel, x::AbstractVector) kernelmatrix_diag!(out, k, x)
@deprecate kerneldiagmatrix!(out::AbstractVector, k::Kernel, x::AbstractVector, y::AbstractVector) kernelmatrix_diag!(out, k, x, y)

@deprecate kerneldiagmatrix(k::Kernel, X::AbstractMatrix; obsdim::Int=defaultobs) kernelmatrix_diag(
    k, X; obsdim=obsdim
)
@deprecate kerneldiagmatrix(k::Kernel, X::AbstractMatrix, Y::AbstractMatrix; obsdim::Int=defaultobs) kernelmatrix_diag(
    k, X, Y; obsdim=obsdim
)
@deprecate kerneldiagmatrix!(out::AbstractVector, k::Kernel, X::AbstractMatrix; obsdim::Int=defaultobs) kernelmatrix_diag!(
    out, k, X; obsdim=obsdim
)
@deprecate kerneldiagmatrix!(out::AbstractVector, k::Kernel, X::AbstractMatrix, Y::AbstractMatrix; obsdim::Int=defaultobs) kernelmatrix_diag!(
    out, k, X, Y; obsdim=obsdim
)
```